libcraft/core/src/dimension.rs

```diff
@@ -0,0 +1,68 @@
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "&'static str")]
+pub enum Dimension {
+    Overworld,
+    TheNether,
+    TheEnd,
+}
+
+impl Dimension {
+    pub fn dim_id(&self) -> i32 {
+        match self {
+            Self::Overworld => 0,
+            Self::TheNether => -1,
+            Self::TheEnd => 1,
+        }
+    }
+
+    pub fn from_dim_id(id: i32) -> Option<Self> {
+        match id {
+            0 => Some(Self::Overworld),
+            -1 => Some(Self::TheNether),
+            1 => Some(Self::TheEnd),
+            _ => None,
+        }
+    }
+
+    pub fn namespaced_id(&self) -> &'static str {
+        match self {
+            Self::Overworld => "minecraft:overworld",
+            Self::TheNether => "minecraft:the_nether",
+            Self::TheEnd => "minecraft:the_end",
+        }
+    }
+
+    pub fn from_namespaced_id(id: &String) -> Option<Self> {
+        match id.as_str() {
+            "minecraft:overworld" => Some(Self::Overworld),
+            "minecraft:the_nether" => Some(Self::TheNether),
+            "minecraft:the_end" => Some(Self::TheEnd),
+            _ => None,
+        }
+    }
+}
+
+impl TryFrom<String> for Dimension {
+    type Error = String;
+
+    fn try_from(namespaced_value: String) -> Result<Self, Self::Error> {
+        if let Some(val) = Self::from_namespaced_id(&namespaced_value) {
+            Ok(val)
+        } else {
+            Err(format!(
+                "There is no dimension with namespaced_id {}",
+                namespaced_value
+            )
+            .to_string())
+        }
+    }
+}
+
+impl Into<&'static str> for Dimension {
```

Comment:
Prefer `From`

Reply:
Same as thing as I commented in `TryFrom`?

Reply:
No its not the same, he should implement `From<Dimension> for &'static str`, this will automaticlly implement `Into<&'static str> for Dimension`

Reply:
Oh didn't know that. So implementing `From` does implement `Into` automatically?

Reply:
One should always prefer implementing `From` over `Into` because implementing `From` automatically provides one with an implementation of `Into` thanks to the blanket implementation in the standard library.

https://doc.rust-lang.org/stable/std/convert/trait.From.html

Reply:
Okay, thanks! 😄